web/profiles/joinup/src/Plugin/Block/AddContentBlock.php

```diff
@@ -18,12 +22,55 @@
  *  admin_label = @Translation("Add content"),
  * )
  */
-class AddContentBlock extends BlockBase {
+class AddContentBlock extends BlockBase
+{
+
+  /**
+   * The collection to join.
+   *
+   * @var \Drupal\rdf_entity\RdfInterface
+   */
+  protected $collection;
+
+  /**
+   * The current route match service.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  protected $currentRouteMatch;
+
+  /**
+   * @var \Drupal\user\UserInterface
+   */
+  protected $account;
+
+  /**
+   * Constructs a AddContentBlock object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param string $plugin_definition
+   *   The plugin implementation definition.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition)
+  {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->currentRouteMatch = \Drupal::routeMatch();
+
+    // @todo: This should be restricted to collection rdf_entities only.
+    // Retrieve the collection from the route.
+    $this->collection = $this->currentRouteMatch->getParameter('rdf_entity');
```

Comment:
Just saw that the @todo is still there. If the parameter is going to contain any RDF entity, then change the name from `$this->collection` to `$this->rdfEntity`.